From 52826858bf8b782463b090819f8ca2d9e0d56449 Mon Sep 17 00:00:00 2001
From: Hideyuki Nagase <hideyukn@microsoft.com>
Date: Thu, 11 Jun 2026 16:02:20 +0000
Subject: [PATCH] Fix leaking syncobj

---
 drivers/hv/dxgkrnl/dxgsyncfile.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/hv/dxgkrnl/dxgsyncfile.c b/drivers/hv/dxgkrnl/dxgsyncfile.c
index 4cd6ee47f7823..587be6fbee8a9 100644
--- a/drivers/hv/dxgkrnl/dxgsyncfile.c
+++ b/drivers/hv/dxgkrnl/dxgsyncfile.c
@@ -297,9 +297,14 @@ int dxgkio_open_syncobj_from_syncfile(struct dxgprocess *process,
 cleanup:
 	if (dmafence)
 		dma_fence_put(dmafence);
-	if (ret) {
-		if (syncobj) 
+	if (syncobj) {
+		if (ret) {
 			dxgsyncobject_destroy(process, syncobj);
+			if (openargs.sync_object.v)
+				dxgvmb_send_destroy_sync_object(process, openargs.sync_object);
+		} else {
+			kref_put(&syncobj->syncobj_kref, dxgsyncobject_release);
+		}
 	}
 	if (adapter)
 		dxgadapter_release_lock_shared(adapter);
-- 
2.50.1

